src/metabase/api/card.clj

```diff
@@ -470,11 +486,11 @@
                                                              :query             dataset_query
                                                              :metadata          result_metadata
                                                              :original-metadata (:result_metadata card-before-update)
-                                                             :dataset?          (if (some? dataset)
-                                                                                  dataset
-                                                                                  (:dataset card-before-update))})
+                                                             :dataset?          (if (some? maybe-turn-to-datset)
+                                                                                  maybe-turn-to-datset
+                                                                                  (card/is-model? card-before-update))})
```

Comment:
This change contains a bug. If we are changing `type` from `"model"` to `"question"`, and `dataset=nil`, then `maybe-turn-to-dataset=nil`. That means the value of `dataset?` here will be `true` but it should be `false`.